admin/create-theme/theme-templates.php

```diff
@@ -113,78 +161,127 @@ public static function add_templates_to_local( $export_type ) {
+				$template->content
+			);
 
-				// If there are external images, add it as a pattern
-				$pattern                 = Theme_Patterns::pattern_from_template( $template_data );
-				$pattern_link_attributes = array(
-					'slug' => $pattern['slug'],
-				);
-				$template_data->content  = Theme_Patterns::create_pattern_link( $pattern_link_attributes );
+			// Write the media assets if there are any
+			if ( $template->media ) {
+				Theme_Media::add_media_to_local( $template->media );
+			}
 
-				// Write the pattern
+			// Write the pattern if it exists
+			if ( $template->pattern ) {
 				file_put_contents(
-					get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' . DIRECTORY_SEPARATOR . $template_data->slug . '.php',
-					$pattern['content']
+					get_stylesheet_directory() . DIRECTORY_SEPARATOR . 'patterns' . DIRECTORY_SEPARATOR . $template->slug . '.php',
+					$template->pattern
 				);
 			}
+		}
+	}
 
-			// Write the template content
-			file_put_contents(
-				get_stylesheet_directory() . DIRECTORY_SEPARATOR . $template_folders['wp_template_part'] . DIRECTORY_SEPARATOR . $template_data->slug . '.html',
-				$template_data->content
-			);
+	public static function escape_text_in_template( $template ) {
+		$template_blocks = parse_blocks( $template->content );
+		foreach ( $template_blocks as &$block ) {
+			$block = self::escape_text_in_block( $block );
+		}
+		$template->content = serialize_blocks( $template_blocks );
+		return $template;
+	}
+
+	public static function escape_text_in_block( $block ) {
+
+		// TODO: More blocks besides just the paragraph block
+		// Can we make it generic?
+		if ( 'core/paragraph' === $block['blockName'] ) {
+			$block = self::escape_paragraph( $block );
+		}
+
+		if ( ! empty( $block['innerBlocks'] ) ) {
+			foreach ( $block['innerBlocks'] as &$inner_block ) {
+				$inner_block = self::escape_text_in_block( $inner_block );
+			}
+		}
+		return $block;
+	}
+
+	/**
+	 * Escape the text in a paragraph block.
+	 */
+	public static function escape_paragraph( $block ) {
+		//TODO: This seems messy.  How can we make this more elegant?
+		$content = $block['innerContent'][0];
+		$doc     = new DOMDocument();
+		$doc->loadHTML( $content, LIBXML_HTML_NOIMPLIED | LIBXML_HTML_NODEFDTD );
+		$elements = $doc->getElementsByTagName( '*' );
+		foreach ( $elements as $element ) {
+			$element->nodeValue = self::escape_text( $element->nodeValue );
+		}
+		$block['innerContent'][0] = html_entity_decode( $doc->saveHTML() );
+		return $block;
+	}
 
-			// Write the media assets
-			Theme_Media::add_media_to_local( $template_data->media );
+	public static function escape_text( $text ) {
+		if ( ! $text ) {
+			return $text;
 		}
+		return "<?php echo __('" . $text . "', '" . wp_get_theme()->get( 'TextDomain' ) . "');?>";
+	}
+
+	public static function eliminate_environment_specific_content( $template ) {
```

Comment:
I think the logic you are working on [here](https://github.com/WordPress/create-block-theme/pull/524/files) would drop in nicely here.

Reply:
That logic has been added.  I'm going to close that PR since the logic is included here.  If anything is lacking as of now we can iterate on this branch or trunk.